File: Messenger/Messenger/Messenger/Models/Match.cs
namespace Messenger.Models
{
    public class Match : Entity
    {
        public User UserOfferer { get; set; }
        public string UserOffererId { get; set; }
        public User RoomOfferer { get; set; }
        public string RoomOffererId { get; set; }
        public int? RoomId { get; set; }

        public string GetReceiverId(string senderId)
        {
            if (senderId == UserOffererId)
                return RoomOffererId;
            if (senderId == RoomOffererId)
                return UserOffererId;
            return null;
        }
    }
}

File: Recommendation/Recommendation/Recommendation/Models/Match.py
from dataclasses import dataclass
from DatabaseContext import db
from Models.User import *
from Models.Room import *

@dataclass
class Match(db.Model):
    __tablename__ = 'Matches'
    Id: int = db.Column(db.Integer, primary_key=True)
    UserOffererId: str = db.Column(db.String, db.ForeignKey(User.Id))
    RoomOffererId: str = db.Column(db.String, db.ForeignKey(User.Id))
    RoomId: int = db.Column(db.Integer, db.ForeignKey(Room.Id), nullable=True)
    FilterType: int = db.Column(db.Integer)
